int32 base_mode
float64 base_angle
int32 lower_mode
float64 lower_angle
int32 upper_mode
float64 upper_angle
